app/controllers/changeActivity/PackagingSiteDetailsController.scala

```diff
@@ -67,18 +68,28 @@ class PackagingSiteDetailsController @Inject()(
 
   def onSubmit(mode: Mode): Action[AnyContent] = controllerActions.withRequiredJourneyData(ChangeActivity).async {
     implicit request =>
-
       val siteList: SummaryList = SummaryListViewModel(
         rows = PackagingSiteDetailsSummary.row2(request.userAnswers.packagingSiteList)
       )
       form.bindFromRequest().fold(
         formWithErrors =>
           Future.successful(BadRequest(view(formWithErrors, mode, siteList))),
 
-        value => {
-          val updatedAnswers = request.userAnswers.set(PackagingSiteDetailsPage, value)
-          updateDatabaseAndRedirect(updatedAnswers, PackagingSiteDetailsPage, mode)
-        }
+        value =>
+          for {
+            updatedAnswers <- Future.fromTry(request.userAnswers.set(PackagingSiteDetailsPage, value))
+            _ <- updateDatabaseWithoutRedirect(updatedAnswers, PackagingSiteDetailsPage)
+            onwardUrl <- if (value) {
+              updateDatabaseWithoutRedirect(updatedAnswers, PackagingSiteDetailsPage).flatMap(_ =>
```

Comment:
Why is the database being updated twice?

Reply:
Also if the update fails no error the journey is continuing

Reply:
2nd update to database removed - duplicate not needed.